R/read_data.R

```diff
@@ -56,18 +49,24 @@ read_data.default <- function(ds) {
   }
 
   # Download the file
-
-
   withr::with_envvar(
     new = c("no_proxy" = "dam-api.bfs.admin.ch"),
-    code = download.file(url = ds$read_path, destfile = temp_file, method = download_method, mode = "wb")
+    code = download.file(
+      url = ds$download_url,
+      destfile = temp_file,
+      method = download_method,
+      mode = "wb"
+    )
   )
   # Import the data
-  ds$data <- rio::import(temp_file, which = ds$sheet, header = TRUE)
+  if (ds$format == "xlsx") {
+    ds$data <- rio::import(temp_file)
+  } else {
+    ds$data <- rio::import(temp_file, which = ds$sheet, header = TRUE)
```

Comment:
If the `sheet` field was removed from the dataset table, how is it derived now?
I assume it is derived for some xls datasets

Reply:
The `sheet` was not really needed. You can take a look here: https://github.com/statistikZH/statbotData/blob/main/data/const/statbot_input_data.csv

It contained mostly the `px_id` and for one dataset a text `data` : these were not needed as there is already another property `px_id` that contains the `px_id`.

Reply:
Makes sense, but then line 65 above still uses `ds$sheet`. Shouldn't that be removed?